solutions/tests/README.md

```diff
@@ -1 +1,174 @@
-# Tests
+# Structured Test Solution and Key Testing Standards
+
+## Overview
+
+This document outlines the structure, guidelines, and standards for the test
+ solutions. It serves to guide developers, quality assurance engineers, and
+  other stakeholders in understanding how to write, execute, and maintain tests
+   within the project.
+
+---
+
+## Table of Contents
+
+1. [Test Structure](#test-structure)
+2. [Testing Guidelines](#testing-guidelines)
+3. [Test Standards](#test-standards)
+4. [Test Coverage](#test-coverage)
+5. [Running Tests](#running-tests)
+
+---
+
+## Test Structure
+
+The test solutions are organized into the following directories:
+
+- **Unit Tests**:
+  - Located in the `/unit` directory.
+  - **Purpose**: Validates individual functions or methods in isolation,
+   ensuring they perform as expected.
+  - **Guideline**: Use mocking to isolate functions and avoid testing dependencies.
+
+- **Integration Tests**:
+  - Located in the `/integration` directory.
+  - **Purpose**: Verifies the interaction between components to ensure they work
+  together as expected.
+
+- **Functional Tests**:
+  - Located in the `/functional` directory.
+  - **Purpose**: Focuses on testing application functionality from the user’s perspective.
+
+- **End-to-End (E2E) Tests**:
+  - Located in the `/e2e` directory.
+  - **Purpose**: Simulates real-world user scenarios and ensures the
+   application’s workflow functions as intended.
+
+- **Fixtures**:
+  - Located in the `/fixtures` directory.
+  - **Purpose**: Contains predefined data, settings, or mock objects for
+   consistency across tests.
+
+### Naming Conventions
+
+- **Test files**: Should follow the pattern `test_<module_name>.py`.
+- **Test functions**: Should be named `test_<functionality>`, e.g., `test_login`.
+
+---
+
+## Testing Guidelines
+
+### Writing Tests
+
+- **Isolate Tests**: Each test should be independent and should not rely on
+   shared states from other tests. This prevents flakiness.
+  
+- **Clear Assertions**: Ensure each test has clear and specific assertions to
+   check expected outcomes.
+  
+- **Mocking**: Use tools like `unittest.mock` or `pytest-mock` to mock external
```

Comment:
We are not supposed to use Mocking in the unittests that are required for this project.